tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -359,29 +352,37 @@ ROOT::Experimental::RNTupleJoinProcessor::RNTupleJoinProcessor(const RNTupleOpen
 
    fNEntries = fPageSource->GetNEntries();
 
-   if (!model)
-      model = fPageSource->GetSharedDescriptorGuard()->CreateModel();
+   if (models.empty())
+      fModel = fPageSource->GetSharedDescriptorGuard()->CreateModel();
+   else
+      fModel = models[0]->Clone();
 
-   fModel = model->Clone();
    fModel->Freeze();
    fEntry = fModel->CreateEntry();
 
    for (const auto &value : *fEntry) {
       auto &field = value.GetField();
       const auto &fieldName = field.GetQualifiedFieldName();
 
-      // If the model has a default entry, use the value pointers from the default entry of the model that was passed to
-      // this constructor. This way, the pointers returned by RNTupleModel::MakeField can be used in the processor loop
-      // to access the corresponding field values.
-      if (!fModel->IsBare()) {
-         auto valuePtr = model->GetDefaultEntry().GetPtr<void>(fieldName);
+      // If the model provided by the user has a default entry, use the value pointers from the default entry of the
+      // model that was passed to this constructor. This way, the pointers returned by RNTupleModel::MakeField can be
+      // used in the processor loop to access the corresponding field values.
+      if (!models.empty() && !models[0]->IsBare()) {
+         auto valuePtr = models[0]->GetDefaultEntry().GetPtr<void>(fieldName);
          fEntry->BindValue(fieldName, valuePtr);
       }
 
       const auto &[fieldContext, _] =
          fFieldContexts.try_emplace(fieldName, field.Clone(fieldName), fEntry->GetToken(fieldName));
       ConnectField(fieldContext->second, *fPageSource, *fEntry);
    }
+
+   for (unsigned i = 0; i < auxNTuples.size(); ++i) {
+      if (models.empty())
+         AddAuxiliary(auxNTuples[i], joinFields);
+      else
+         AddAuxiliary(auxNTuples[i], joinFields, std::move(models[i + 1]));
```

Comment:
This requires `models.size() > auxNTuples.size()` is it checked somewhere in the routine and/or is it already guaranteed by the control flow (`models` and `auxNTuples` are parameters of this function)?

Reply:
It is checked in `CreateJoin` (l. 124) and this constructor is private, so unless I'm overlooking something it should be guaranteed by the workflow.